Fix flaky test_debug_info.

pritamdamania87 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#31675 Fix flaky test_debug_info.**

This test could be flaky since there could be inflight RPC requests as
part of startup which might not have finished. As a result, if they finish
between the different calls to retrieve debug_info, there could be a problem
since we would report separate information. As a result, we wait to ensure
the metrics stabilize to avoid flakiness.

Differential Revision: [D19242588](https://our.internmc.facebook.com/intern/diff/D19242588/)

pritamdamania committed:
Fix flaky test_debug_info.

This test could be flaky since there could be inflight RPC requests as
part of startup which might not have finished. As a result, if they finish
between the different calls to retrieve debug_info, there could be a problem
since we would report separate information. As a result, we wait to ensure
the metrics stabilize to avoid flakiness.

Differential Revision: [D19242588](https://our.internmc.facebook.com/intern/diff/D19242588/)

[ghstack-poisoned]

mrshenli commented on test/rpc_test.py:
```diff
@@ -1297,27 +1297,37 @@ def test_local_shutdown(self):
+            expected = self._get_expected_debug_info()
+            time.sleep(1)
+
         self.assertEqual(expected, info)
```
Thanks for fixing! I was intended to only test keys in this test case. Will it be sufficient to avoid the flakiness if we compare `expected.keys()` and `info.keys()`?

pritamdamania87 replied:
I see, yes if we compare keys only then we don't need the additional changes. Let me update the PR.

pritamdamania committed:
Update on "Fix flaky test_debug_info."


This test could be flaky since there could be inflight RPC requests as
part of startup which might not have finished. As a result, if they finish
between the different calls to retrieve debug_info, there could be a problem
since we would report separate information. As a result, we wait to ensure
the metrics stabilize to avoid flakiness.

Differential Revision: [D19242588](https://our.internmc.facebook.com/intern/diff/D19242588/)

[ghstack-poisoned]

mrshenli reviewed (approved):
Thanks for fixing!!

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 0f2dd82a:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=0f2dd82aa71066c439597a552bd2167aa2a46d26).

### :detective: 1 new failure recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4087484) caffe2_onnx_py3_6_clang7_ubuntu16_04_test (1/1)
**Step:** "Test" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=78780469) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=78780469))
<details>
<summary>
<code>ore dumped) pytest "${args[@]}" -k 'not (TestOperators and test_full_like) and not (TestOperators and test_zeros_like) and not (TestOperators and test_ones_like) and not (TestModels and test_vgg16) and not (TestModels and test_vgg16_bn) and not (TestModels and test_vgg19) and not (TestModels and test_vgg19_bn)' --ignore "$top_dir/test/onnx/test_pytorch_onnx_onnxruntime.py" --ignore "$top_dir/test/onnx/test_custom_ops.py" --ignore "$top_dir/test/onnx/test_models_onnxruntime.py" "${test_paths[@]}"</code>
</summary>

```
Dec 30 21:10:57   File "/usr/local/lib/python3.6/dist-packages/pluggy/hooks.py", line 286 in __call__ 
Dec 30 21:10:57   File "/usr/local/lib/python3.6/dist-packages/_pytest/main.py", line 237 in _main 
Dec 30 21:10:57   File "/usr/local/lib/python3.6/dist-packages/_pytest/main.py", line 193 in wrap_session 
Dec 30 21:10:57   File "/usr/local/lib/python3.6/dist-packages/_pytest/main.py", line 230 in pytest_cmdline_main 
Dec 30 21:10:57   File "/usr/local/lib/python3.6/dist-packages/pluggy/callers.py", line 187 in _multicall 
Dec 30 21:10:57   File "/usr/local/lib/python3.6/dist-packages/pluggy/manager.py", line 86 in <lambda> 
Dec 30 21:10:57   File "/usr/local/lib/python3.6/dist-packages/pluggy/manager.py", line 92 in _hookexec 
Dec 30 21:10:57   File "/usr/local/lib/python3.6/dist-packages/pluggy/hooks.py", line 286 in __call__ 
Dec 30 21:10:57   File "/usr/local/lib/python3.6/dist-packages/_pytest/config/__init__.py", line 90 in main 
Dec 30 21:10:57   File "/usr/local/bin/pytest", line 11 in <module> 
re dumped) pytest "${args[@]}" -k 'not (TestOperators and test_full_like) and not (TestOperators and test_zeros_like) and not (TestOperators and test_ones_like) and not (TestModels and test_vgg16) and not (TestModels and test_vgg16_bn) and not (TestModels and test_vgg19) and not (TestModels and test_vgg19_bn)' --ignore "$top_dir/test/onnx/test_pytorch_onnx_onnxruntime.py" --ignore "$top_dir/test/onnx/test_custom_ops.py" --ignore "$top_dir/test/onnx/test_models_onnxruntime.py" "${test_paths[@]}" 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>
